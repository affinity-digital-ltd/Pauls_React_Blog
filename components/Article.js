import { Component } from 'react'
import gql from 'graphql-tag'

import { MarkdownPreview } from 'react-marked-markdown'
import moment from 'moment'

import UserComments from '../components/UserComments'

class Article extends Component {
  render () {
    const { article } = this.props

    return <article className='o-main-container'>
      <div className='c-article'>
        <h2 className='c-article__header'>{article.title}</h2>
        <div className='c-header-divider' />
        <h3 className='c-article__author'>Article by: {article.author}</h3>
        <time className='c-article__published'><p>Published on: {moment(article.createdAt).format('ll')}</p></time>
        <figure>
          <img className='c-article__main-img' src={`http://localhost:3001${article.image}`} />
        </figure>
        <div className='c-article__text'>
          <MarkdownPreview value={article.text} />
        </div>
      </div>
      {/* <UserComments comments={article.comments} /> */}
      <a className='c-user-comments__back-btn'>Back to Article List</a>
    </article>
  }
}

export default Article
