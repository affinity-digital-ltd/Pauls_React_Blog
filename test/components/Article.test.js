import Article from '../../components/Article'

describe('<Article />', () => {
  it('it renders correctly', () => {
    const article = {
      'author': 'Paul Brighton',
      'title': 'New Article #3',
      'text': 'A third article!!',
      'createdAt': '2018-12-12 10:20:46 UTC',
      'id': '3',
      'image': '/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg'
    }

    const component = shallow(
      <Article article={article} />
    )

    expect(component).toMatchSnapshot()
  })
})
