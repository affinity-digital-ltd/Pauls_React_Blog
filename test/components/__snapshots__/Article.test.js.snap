// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    article={
      Object {
        "author": "Paul Brighton",
        "createdAt": "2018-12-12 10:20:46 UTC",
        "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
        "text": "A third article!!",
        "title": "New Article #3",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-article"
        >
          <h2
            className="c-article__header"
          >
            New Article #3
          </h2>
          <div
            className="c-header-divider"
          />
          <h3
            className="c-article__author"
          >
            Article by: 
            Paul Brighton
          </h3>
          <time
            className="c-article__published"
          >
            <p>
              Published on: 
              2018-12-12 10:20:46 UTC
            </p>
          </time>
          <figure>
            <img
              className="c-article__main-img"
              src="https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg"
            />
          </figure>
          <div
            className="c-article__text"
          >
            <MarkdownPreview
              value="A third article!!"
            />
          </div>
        </div>,
        <div
          className="c-user-comments__articles-link"
        >
          <Link
            href="/"
          >
            <a
              className="c-user-comments__back-btn"
            >
              Back to Article List
            </a>
          </Link>
        </div>,
      ],
      "className": "o-main-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="c-article__header"
            >
              New Article #3
            </h2>,
            <div
              className="c-header-divider"
            />,
            <h3
              className="c-article__author"
            >
              Article by: 
              Paul Brighton
            </h3>,
            <time
              className="c-article__published"
            >
              <p>
                Published on: 
                2018-12-12 10:20:46 UTC
              </p>
            </time>,
            <figure>
              <img
                className="c-article__main-img"
                src="https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg"
              />
            </figure>,
            <div
              className="c-article__text"
            >
              <MarkdownPreview
                value="A third article!!"
              />
            </div>,
          ],
          "className": "c-article",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New Article #3",
              "className": "c-article__header",
            },
            "ref": null,
            "rendered": "New Article #3",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-header-divider",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Article by: ",
                "Paul Brighton",
              ],
              "className": "c-article__author",
            },
            "ref": null,
            "rendered": Array [
              "Article by: ",
              "Paul Brighton",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Published on: 
                2018-12-12 10:20:46 UTC
              </p>,
              "className": "c-article__published",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Published on: ",
                  "2018-12-12 10:20:46 UTC",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Published on: ",
                "2018-12-12 10:20:46 UTC",
              ],
              "type": "p",
            },
            "type": "time",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="c-article__main-img"
                src="https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-article__main-img",
                "src": "https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "figure",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MarkdownPreview
                value="A third article!!"
              />,
              "className": "c-article__text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "value": "A third article!!",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            href="/"
          >
            <a
              className="c-user-comments__back-btn"
            >
              Back to Article List
            </a>
          </Link>,
          "className": "c-user-comments__articles-link",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a
              className="c-user-comments__back-btn"
            >
              Back to Article List
            </a>,
            "href": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back to Article List",
              "className": "c-user-comments__back-btn",
            },
            "ref": null,
            "rendered": "Back to Article List",
            "type": "a",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-article"
          >
            <h2
              className="c-article__header"
            >
              New Article #3
            </h2>
            <div
              className="c-header-divider"
            />
            <h3
              className="c-article__author"
            >
              Article by: 
              Paul Brighton
            </h3>
            <time
              className="c-article__published"
            >
              <p>
                Published on: 
                2018-12-12 10:20:46 UTC
              </p>
            </time>
            <figure>
              <img
                className="c-article__main-img"
                src="https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg"
              />
            </figure>
            <div
              className="c-article__text"
            >
              <MarkdownPreview
                value="A third article!!"
              />
            </div>
          </div>,
          <div
            className="c-user-comments__articles-link"
          >
            <Link
              href="/"
            >
              <a
                className="c-user-comments__back-btn"
              >
                Back to Article List
              </a>
            </Link>
          </div>,
        ],
        "className": "o-main-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="c-article__header"
              >
                New Article #3
              </h2>,
              <div
                className="c-header-divider"
              />,
              <h3
                className="c-article__author"
              >
                Article by: 
                Paul Brighton
              </h3>,
              <time
                className="c-article__published"
              >
                <p>
                  Published on: 
                  2018-12-12 10:20:46 UTC
                </p>
              </time>,
              <figure>
                <img
                  className="c-article__main-img"
                  src="https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg"
                />
              </figure>,
              <div
                className="c-article__text"
              >
                <MarkdownPreview
                  value="A third article!!"
                />
              </div>,
            ],
            "className": "c-article",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New Article #3",
                "className": "c-article__header",
              },
              "ref": null,
              "rendered": "New Article #3",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-header-divider",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Article by: ",
                  "Paul Brighton",
                ],
                "className": "c-article__author",
              },
              "ref": null,
              "rendered": Array [
                "Article by: ",
                "Paul Brighton",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Published on: 
                  2018-12-12 10:20:46 UTC
                </p>,
                "className": "c-article__published",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Published on: ",
                    "2018-12-12 10:20:46 UTC",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Published on: ",
                  "2018-12-12 10:20:46 UTC",
                ],
                "type": "p",
              },
              "type": "time",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className="c-article__main-img"
                  src="https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "c-article__main-img",
                  "src": "https://pauls-blog.herokuapp.com//rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "figure",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MarkdownPreview
                  value="A third article!!"
                />,
                "className": "c-article__text",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "value": "A third article!!",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              href="/"
            >
              <a
                className="c-user-comments__back-btn"
              >
                Back to Article List
              </a>
            </Link>,
            "className": "c-user-comments__articles-link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-user-comments__back-btn"
              >
                Back to Article List
              </a>,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Back to Article List",
                "className": "c-user-comments__back-btn",
              },
              "ref": null,
              "rendered": "Back to Article List",
              "type": "a",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
